GANDLF/compute/forward_pass.py

```diff
@@ -458,23 +460,20 @@ def validate_network(
                 if ((batch_idx + 1) % (len(valid_dataloader) / 2) == 0) and (
                     (batch_idx + 1) < len(valid_dataloader)
                 ):
-                    print(
+                    logger.debug(
                         "\nHalf-Epoch Average " + mode + " loss : ",
                         total_epoch_valid_loss / (batch_idx + 1),
                     )
-                    for metric in params["metrics"]:
-                        if isinstance(total_epoch_valid_metric[metric], np.ndarray):
-                            to_print = (
-                                total_epoch_valid_metric[metric] / (batch_idx + 1)
-                            ).tolist()
-                        else:
-                            to_print = total_epoch_valid_metric[metric] / (
-                                batch_idx + 1
-                            )
-                        print(
-                            "Half-Epoch Average " + mode + " " + metric + " : ",
-                            to_print,
-                        )
+                for metric in params["metrics"]:
+                    if isinstance(total_epoch_valid_metric[metric], np.ndarray):
+                        to_print = (
+                            total_epoch_valid_metric[metric] / (batch_idx + 1)
+                        ).tolist()
+                    else:
+                        to_print = total_epoch_valid_metric[metric] / (batch_idx + 1)
+                    logger.debug(
+                        "Half-Epoch Average " + mode + " " + metric + " : ", to_print
+                    )
```

Comment:
```suggestion
                  for metric in params["metrics"]:
                      if isinstance(total_epoch_valid_metric[metric], np.ndarray):
                          to_print = (
                              total_epoch_valid_metric[metric] / (batch_idx + 1)
                          ).tolist()
                      else:
                          to_print = total_epoch_valid_metric[metric] / (batch_idx + 1)
                      logger.debug(
                          "Half-Epoch Average " + mode + " " + metric + " : ", to_print
                      )
```
You occasionally shifted this piece of code out from `if half-epoch` condition